testing/e2e/clients/aks.go

```diff
@@ -76,6 +91,37 @@ var OsmClusterOpt = McOpt{
 	},
 }
 
+var ServicePrincipalClusterOpt = McOpt{
+	Name: "service principal cluster",
+	fn: func(mc *armcontainerservice.ManagedCluster, opt McOptFields) error {
```

Comment:
I thought you were gonna take the approach of having a mcOpt generator that would use the external argument and return a function that used those? so in this case spOpts would just be passed into the mcOpt generator?

Reply:
that was the plan, but we need access to the service principal fields in the `provision` function, which calls the McOpt.fn function, so i just passed it there to remove the extra layer in the call stack. Do you think adding a constructor for it would be cleaner?
i ended up passing the appObjectId as a flag instead of an environment var which made me question the constructor approach